web/backend/src/Server.ts

```diff
@@ -216,10 +216,23 @@ app.post('/api/add_role', (req, res) => {
   const { sciper } = req.body;
   const { role } = req.body;
 
-  usersDB.put(sciper, role).catch((error) => {
-    res.status(500).send('Failed to add role');
-    console.log(error);
-  });
+  // The sciper has to contain 6 numbers
+  if (sciper < 999999 && sciper > 100000) {
+    // call https://search-api.epfl.ch/api/ldap?q=228271 if the answer
+    // empty then sciper invalid
+    axios.get(`https://search-api.epfl.ch/api/ldap?q=${sciper}`).then((response) => {
```

Comment:
what if the `get` fails ? it seems you don't catch that case.

Reply:
Yes I missed this one, I just added a catch in a new commit